### ISSUE TITLE
Device::create_buffer is sometimes slow (4ms) and slows down rendering.

### ISSUE THREAD

**Issue by John-Nagle** [gfx-rs/wgpu#5984]
**Description**
Performance bottleneck found with Tracy: too much time is spent in "Device::create_buffer" and that seems to delay other thread.
This should be a fast operation, but it's taking about 4ms at times.

**Repro steps**
1. Get [render-bench.](https://github.com/John-Nagle/render-bench). Build branch "hp" in release mode.
2. Run with the Tracy profiler 0.10
3. Zoom in on the slowest frames in the profiler.

The test creates a large number of visible objects on screen, waits 10 seconds, deletes them, waits 10 seconds, etc. So capture one full create/delete cycle.

**Expected vs observed behavior**
The part of the code in the profiling scope "Device::create_buffer" is 1) taking as long as 4ms, and 2) locking out some other operations in the render thread. As far as I can tell, that ought to be a fast operation.

**Extra materials**
Screenshot of the trace.
![renderbenchcreatebuffer1](https://github.com/user-attachments/assets/d63a90d5-84cc-4068-a992-378662559792)

Full Tracy trace file:
[renderbenchcreatebuffer.zip](https://github.com/user-attachments/files/16288305/renderbenchcreatebuffer.zip)

**Platform**
WGPU 0.20 from crates.io
Linux 22.04 LTS.
NVidia 3070. Driver 535 (proprietary, tested)

**Comment by partisani** [comment]
What do you mean by Linux 22.04 LTS? The latest version of the linux kernel is 6.10

**Comment by John-Nagle** [comment]
Ubuntu 22.04 LTS

**Comment by Wumpf** [comment]
> As far as I can tell, that ought to be a fast operation.

I think wgpu should do a better job documenting that it is in fact known to be a very slow operation.

That said, regardless it needs to be looked at if it really _has_ to lock out render pass recording (or vice versa, not that it matters :)).
Wgpu is making some considerable progress in this area, so might be worth checking if the just released 22.0.0 got better in that regard, but I'd be kinda surprised if it's fundamentally different (but who knows! personally I lost a bit track of all the refactors that went in 😅).

Ideally, it would only be an "occasionally very slow" operation, i.e. whenever it actually happens to bottom out to an allocation in the driver (which shouldn't happen all that often)!

**Comment by John-Nagle** [comment]
> just released (WGPU) 22.0.0

OK, I will upgrade all my code, and Rend3, and re-test. More tomorrow.

> very slow operation.

Indeed. 4ms is slow for something in the main render loop.

**Comment by cwfitzgerald** [comment]
The buffer in question is the vertex buffer (you can tell by it being accessed by the mesh manager). This buffer can get very large, and large allocations can take a second for us to generate, as the underlying memory allocation takes a little bit. It shouldn't block the main thread, however.

**Comment by John-Nagle** [comment]
Waiting for **wgpu-egui** and **wgpu-profiler** to catch up to **wgpu** 22.0.0. Both have the appropriate pull requests.

**Comment by John-Nagle** [comment]
> The buffer in question is the vertex buffer (you can tell by it being accessed by the mesh manager). This buffer can get very large, and large allocations can take a second for us to generate, as the underlying memory allocation takes a little bit. It shouldn't block the main thread, however.

Right. Profiling can show this happening, but extracting cross-thread cause and effect from profiling data is hard.

**Comment by John-Nagle** [comment]
The pull request to fix wgpu-profiling failed. See https://github.com/Wumpf/wgpu-profiler/pull/75

A new WGPU version is needed to fix that, apparently.
